File: Domains/Apps/Workspace/Typescript/Workspace/src/domain/apps/TelecommunicationsNumber.ts
import { TelecommunicationsNumber } from "../generated/TelecommunicationsNumber.g";

declare module "../generated/TelecommunicationsNumber.g" {
  interface TelecommunicationsNumber {
    displayName;
  }
}

Object.defineProperty(TelecommunicationsNumber.prototype, "displayName", {
  get(this: TelecommunicationsNumber) {
    let numberString: string = "";
    if (this.CountryCode || this.AreaCode) {
      if (this.CountryCode && this.AreaCode) {
        numberString = this.CountryCode + " " + this.AreaCode;
      } else if (this.CountryCode) {
        numberString = this.CountryCode;
      } else {
        numberString = this.AreaCode;
      }
    }

    if (numberString === "" && this.ContactNumber) {
      return numberString = this.ContactNumber;
    } else {
      return numberString += " " + this.ContactNumber;
    }
  },
});

File: dotnet/base/database/domain/base/relation/telecommunicationsnumber.cs
// <copyright file="TelecommunicationsNumber.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Domain
{
    public partial class TelecommunicationsNumber
    {
        public bool IsPostalAddress => false;

        public override string ToString()
        {
            var numberString = string.Empty;
            if (this.ExistCountryCode || this.ExistAreaCode)
            {
                if (this.ExistCountryCode && this.ExistAreaCode)
                {
                    numberString = this.CountryCode + ' ' + this.AreaCode;
                }
                else if (this.ExistCountryCode)
                {
                    numberString = this.CountryCode;
                }
                else
                {
                    numberString = this.AreaCode;
                }
            }

            if (string.IsNullOrEmpty(numberString) && this.ExistContactNumber)
            {
                return this.ContactNumber;
            }
            else
            {
                numberString += ' ' + this.ContactNumber;
                return numberString;
            }
        }
    }
}
